src/test/java/com/amazon/opendistroforelasticsearch/sql/esdomain/mapping/FieldMappingTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.esdomain.mapping;
+
+import com.amazon.opendistroforelasticsearch.sql.domain.Field;
+import com.amazon.opendistroforelasticsearch.sql.util.MatcherUtils;
+import org.elasticsearch.action.admin.indices.mapping.get.GetFieldMappingsResponse.FieldMappingMetaData;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyMap;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Unit test for {@code FieldMapping} with trivial methods ignored such as isSpecified, isMetaField etc.
+ */
+public class FieldMappingTest {
+
+    @Test
+    public void testFieldMatchesWildcardPatternSpecifiedInQuery() {
+        assertThat(
+            new FieldMapping("employee.first", typeMappings(), fieldsSpecifiedInQuery("employee.*")),
+            isWildcardSpecified(true)
+        );
+    }
+
+    @Test
+    public void testFieldMismatchesWildcardPatternSpecifiedInQuery() {
+        assertThat(
+            new FieldMapping("employee.first", typeMappings(), fieldsSpecifiedInQuery("manager.*")),
+            isWildcardSpecified(false)
+        );
+    }
+
+    @Test
+    public void testFieldIsProperty() {
+        assertThat(
+            new FieldMapping("employee.first"),
+            isPropertyField(true)
+        );
+
+        assertThat(
+            new FieldMapping("employee.first.keyword"),
+            isPropertyField(true)
+        );
+    }
+
+    @Test
+    public void testFieldIsNotProperty() {
+        assertThat(
```

Comment:
Please extract those sequential assertThat into different tests

Reply:
Good point! Fixed it.